specification/communication/data-plane/Identity/stable/2023-08-01/CommunicationIdentity.json

```diff
@@ -0,0 +1,460 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "CommunicationIdentityClient",
+    "description": "Azure Communication Identity Service",
+    "version": "2023-08-01"
+  },
+  "paths": {
+    "/identities": {
+      "post": {
+        "tags": [
+          "Identity"
+        ],
+        "summary": "Create a new identity, and optionally, an access token.",
+        "operationId": "CommunicationIdentity_Create",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "header",
+            "name": "Repeatability-Request-ID",
+            "description": "Unique identifier of the request.",
+            "type": "string"
+          },
+          {
+            "in": "header",
+            "name": "Repeatability-First-Sent",
+            "description": "Date and time at which the request was first created.",
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "If specified, creates also a Communication Identity access token associated with the identity and containing the requested scopes.",
+            "schema": {
+              "$ref": "#/definitions/CommunicationIdentityCreateRequest"
+            }
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "../../../Common/stable/2021-03-07/common.json#/definitions/CommunicationErrorResponse"
+            }
+          },
+          "201": {
+            "description": "Created - Returns the created identity.",
+            "schema": {
+              "$ref": "#/definitions/CommunicationIdentityAccessTokenResult"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Create an Identity and optionally an access token with optional idempotent processing using Repeatability-Request-ID and Repeatability-First-Sent headers. If the headers are not provided, the request will be processed in a non-idempotent way.": {
+            "$ref": "./examples/CreateIdentity.json"
+          }
+        }
+      }
+    },
+    "/identities/{id}": {
+      "delete": {
+        "tags": [
+          "Identity"
+        ],
+        "summary": "Delete the identity, revoke all tokens for the identity and delete all associated data.",
```

Comment:
there was a discussion about whether we should provide idempotency for the APIs that are idempotent naturally, I was wondering if we agreed to introduce the feature for all APIs?

Reply:
I guess for this question @petrsvihlik can answer :)

Reply:
We'll just make these APIs aware of the repeatability headers and keep the current functionality, following this Azure [guideline](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#repeatability-of-requests):

> When understood, all endpoints co-located behind a DNS name MUST understand the header. This means that a service MUST NOT ignore the presence of a header for any endpoints behind the DNS name, but rather fail the request containing a Repeatability-Request-ID header if that particular endpoint lacks support for repeatable requests. Such partial support SHOULD be avoided due to the confusion it causes for clients.


That means when someone decides to set the headers, they must be set correctly - we'll perform the header validation and continue with the normal, naturally idempotent execution.